Add ASAN flag to compiler when preprocessing sandboxes

pvollan opened the pull request:
#### 65f3f986599dda3b52f25844623f3dbe05a32f75
<pre>
Add ASAN flag to compiler when preprocessing sandboxes
<a href="https://bugs.webkit.org/show_bug.cgi?id=263218">https://bugs.webkit.org/show_bug.cgi?id=263218</a>
rdar://117038358

Reviewed by Elliott Williams.

Add ASAN flag to compiler when preprocessing sandboxes which will make sure ASAN_ENABLED is set.

* Source/WebKit/DerivedSources.make:

Canonical link: <a href="https://commits.webkit.org/269417@main">https://commits.webkit.org/269417@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ab32602e641af8dc1488e5c3642bc84e32ae026

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22498 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/152 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23589 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24403 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20827 "Built successfully") 
| | [💥 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/1174 "An unexpected error occured. Recent messages:Failed to print configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23030 "Build is in progress. Recent messages:OS: Ventura (13.4.1), Xcode: 14.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (warnings)") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21823 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22738 "Passed tests") | [💥 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/49/builds/1174 "An unexpected error occured. Recent messages:Failed to print configuration") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19541 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25267 "Built successfully") | 
| | [💥 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/49/builds/1174 "An unexpected error occured. Recent messages:Failed to print configuration") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/26652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [💥 🧪 api-ios](https://ews-build.webkit.org/#/builders/49/builds/1174 "An unexpected error occured. Recent messages:Failed to print configuration") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20644 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24494 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/87 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/17948 "Passed tests") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/64 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5361 "Built successfully and passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/104 "Build is in progress. Recent messages:OS: Sonoma (14.0), Xcode: 15.0; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running compile-webkit") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/101 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WebKit/DerivedSources.make:
```diff
@@ -399,7 +403,7 @@ all : $(SANDBOX_PROFILES_WITHOUT_WEBPUSHD) $(WEBPUSHD_SANDBOX_PROFILE) $(SANDBOX
 %.sb : %.sb.in
 	@echo Pre-processing $* sandbox profile...
-	grep -o '^[^;]*' $< | $(CC) $(SDK_FLAGS) $(TARGET_TRIPLE_FLAGS) $(SANDBOX_DEFINES) $(TEXT_PREPROCESSOR_FLAGS) $(FRAMEWORK_FLAGS) $(HEADER_FLAGS) $(EXTERNAL_FLAGS) -include "wtf/Platform.h" - > $@
+	grep -o '^[^;]*' $< | $(CC) $(SANITIZE_FLAGS) $(SDK_FLAGS) $(TARGET_TRIPLE_FLAGS) $(SANDBOX_DEFINES) $(TEXT_PREPROCESSOR_FLAGS) $(FRAMEWORK_FLAGS) $(HEADER_FLAGS) $(EXTERNAL_FLAGS) -include "wtf/Platform.h" - > $@
```
Let's add this flag in the definition of `platform_h_compiler_command`, too (L345). Otherwise looks good!

pvollan replied:
Fixed in latest patch.

Thanks for reviewing!

pvollan committed:
Add ASAN flag to compiler when preprocessing sandboxes
https://bugs.webkit.org/show_bug.cgi?id=263218
rdar://117038358

Reviewed by Elliott Williams.

Add ASAN flag to compiler when preprocessing sandboxes which will make sure ASAN_ENABLED is set.

* Source/WebKit/DerivedSources.make:

Canonical link: https://commits.webkit.org/269417@main
